src/main/scala/zio/prelude/DivideShape.scala

```diff
@@ -0,0 +1,49 @@
+package zio.prelude
+
+import zio.prelude.newtypes.{ Prod, Sum }
+
+trait DivideShape[A, +Addition[x] <: Associative[x], +Multiplication[x] <: InverseNonZero[x]]
```

Comment:
I'd like to brainstorm a bit on the names here. This definitely takes it down a certain path. 

In addition, the alternate (type member) encoding could clean things up a bit here, if it ends up working out.

Reply:
Happy to change the naming! Any suggestions?
(I will definitely try the type member encoding again.)